ml/create_model.py

```diff
@@ -208,39 +213,69 @@ def target_getter(df):
         pass
 
 # Setting up features and labels
-features = embeddings
+features =  torch.FloatTensor(embeddings).to(device)
+target = torch.LongTensor(target).to(device)
 
 # Splitting the data
 X_train, X_test, y_train, y_test = train_test_split(features, target, test_size=0.2, random_state=42)
 
-# Param grid for random search
-params = {
-        'n_estimators' : [250, 300, 350, 400, 450, 500],
-        'min_child_weight': [1, 5, 10],
-        'gamma': [0.01, 0.1, 0.5, 1, 1.5],
-        'subsample': [0.6, 0.8, 1.0],
-        'learning_rate': [0.01, 0.1],
-        'max_depth': [2, 3, 4, 5, 6],
-        'random_state': [42]
-        }
-
-# Instantiate and test the model
-print()
-model = xgb.XGBClassifier()
-rs_model = RandomizedSearchCV(model, param_distributions=params, n_iter=5, scoring='accuracy', cv=3, verbose=3)
-rs_model.fit(X_train, y_train)
-y_pred = rs_model.predict(X_test)
-
-# Save model
-with open("ml/model.pkl", "wb") as handle:
-    pickle.dump(rs_model, handle)
-
-# Save encoder
-with open("ml/encoder.pkl", "wb") as handle:
-    pickle.dump(encoder, handle)
+model = nn.Sequential(
+    nn.Linear(X_train.shape[1], 128),
+    nn.ReLU(),
+    nn.Dropout(p=0.2),
+    nn.Linear(128, 24),
+    nn.ReLU(),
+    nn.Dropout(p=0.2),
+    nn.Linear(24, 3)
+).to(device)
+
+loss_function = nn.CrossEntropyLoss()
+optimizer = torch.optim.Adam(model.parameters())
+
+losses = []
+print(" ")
+print("Training model...")
+for epoch in range(750):
+    y_hat = model(X_train)
+    loss = loss_function(y_hat, y_train)
+    losses.append(loss.item())
+
+    model.zero_grad()
+    loss.backward()
+
+    #if epoch % 10 == 0:
+    sys.stdout.write(f'\rLoss of {round(float(loss), 2)} at epoch {epoch} of 750')
```

Comment:
epoch + 1? otherwise it stops at 749 of 750

Reply:
Good point. Changed epoch to epoch + 1